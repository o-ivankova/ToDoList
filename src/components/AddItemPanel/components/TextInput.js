import React, { useState } from "react";

// eslint-disable-next-line react/prop-types
const TextInput = ({ onSubmit }) => {
  const [input, setInput] = useState("");

  const onInputChange = (e) => {
    setInput(e.target.value);
  };

  return (
    <input
      type="text"
      className="form-control"
      placeholder="Type your to do"
      onChange={onInputChange}
      value={input}
      onSubmit={() => onSubmit(input)}
    />
  );
};

export default TextInput;
