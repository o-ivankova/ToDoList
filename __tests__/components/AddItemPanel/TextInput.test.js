import TextInput from "../../../src/components/AddItemPanel/components/TextInput";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { expect, jest, test, describe, afterAll } from "@jest/globals";
import "@testing-library/jest-dom/extend-expect";
import { act } from "react-test-renderer";
import renderer from "react-test-renderer";

describe("AddItemPanel tests", () => {
  const setup = (input) => {
    const onInputChange = jest.fn();
    render(<TextInput onInputChange={onInputChange} input={input} />);
  };

  test("component is rendered", () => {
    setup("");

    expect(screen.getByRole("textbox")).not.toBeNull;
  });

  test("on input the function is called", () => {
    const onInputChange = jest.fn();
    render(<TextInput onInputChange={onInputChange} input="" />);

    act(() => {
      userEvent.type(screen.getByRole("textbox"), "Buy milk");
    });

    expect(onInputChange).toBeCalled;
  });

  test("input value is displayed", () => {
    const userInput = "Buy milk";
    setup(userInput);

    expect(screen.getByRole("textbox").value).toBe(userInput);
  });

  test("snapshot test", () => {
    const onInputChange = jest.fn();
    const domTree = renderer
      .create(
        <TextInput onInputChange={onInputChange} input="" />
      )
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });

  afterAll(() => {
    jest.clearAllMocks();
  });
});
