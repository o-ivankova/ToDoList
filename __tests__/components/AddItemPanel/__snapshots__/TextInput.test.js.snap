// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddItemPanel tests snapshot test 1`] = `
<input
  className="addItemInput"
  data-testid="input"
  onChange={[Function]}
  placeholder="Type your to do"
  type="text"
  value=""
/>
`;
